pygac_fdr/metadata.py

```diff
@@ -138,47 +167,69 @@ def get_metadata(self, filenames):
 
     def save_sql(self, mda, dbfile, if_exists):
         """Save metadata to sqlite database."""
-        con = sqlite3.connect(dbfile)
-        mda.to_sql(name='metadata', con=con, if_exists=if_exists)
-        con.commit()
-        con.close()
+        engine = sqlalchemy.create_engine('sqlite:///{0}'.format(dbfile))
+        mda.to_sql(name="metadata", con=engine, if_exists=if_exists)
 
     def read_sql(self, dbfile):
         """Read metadata from sqlite database."""
-        with sqlite3.connect(dbfile) as con:
-            mda = pd.read_sql('select * from metadata', con)
-        mda = mda.set_index(['platform', 'level_1'])
-        mda.fillna(value=np.nan, inplace=True)
-        for col in mda.columns:
-            if 'time' in col:
-                mda[col] = mda[col].astype('datetime64[ns]')
```

Comment:
Does the new solution convert time columns in the dataframe to `datetime64` and replace NULL items with NaN? I don't remember why, but I think this was needed for the overlap computation to work correctly.

Reply:
That's a nice use case for the system tests actually

Reply:
I was a bit surprised by these lines and did a test and it worked, meaning the types were conserved when writing and re-reading into the DataFrame.

Reply:
Magic! Curious how the system tests will behave.